### PR TITLE
Restore native backspace behaviour

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Restores a part of #11287. It should not restore any bugginess! :)

We need to make sure none of the issues that #10799 fixed are reintroduced.

Still needs e2e test and investigate failing e2e tests.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Comment by iseulde** [comment]
There's one failing e2e test: if you select a whole paragraph, it seems TinyMCE is doing some weird stuff.

File: packages/editor/src/components/rich-text/tinymce.js [hunk 1]
```diff
@@ -252,9 +252,11 @@ export default class TinyMCE extends Component {
 
 	onKeyDown( event ) {
 		const { keyCode } = event;
+		const { startContainer, startOffset, endContainer, endOffset } = getSelection().getRangeAt( 0 );
+		const isCollapsed = startContainer === endContainer && startOffset === endOffset;
```
Comment (iseulde) [inline packages/editor/src/components/rich-text/tinymce.js]: `getSelection().isCollapsed` is not accurate, should prob be commented.

File: packages/editor/src/components/rich-text/index.js [hunk 2]
```diff
@@ -549,15 +545,18 @@ export class RichText extends Component {
 		const { keyCode } = event;
 
 		if ( keyCode === DELETE || keyCode === BACKSPACE ) {
-			if ( this.onDeleteKeyDown( event ) ) {
+			const value = this.createRecord();
+			const start = getSelectionStart( value );
+			const end = getSelectionEnd( value );
+
+			// Always handle uncollapsed selections ourselves.
```
Comment (iseulde) [inline packages/editor/src/components/rich-text/index.js]: Could use more comment sorry.

**Comment by notnownikki** [comment]
Sure :)

File: test/e2e/specs/writing-flow.test.js [hunk 3]
```diff
@@ -223,4 +223,22 @@ describe( 'adding blocks', () => {
...
+	} );
+
+	it( 'should not delete trailing spaces when deleting a word with alt + backspace', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'alpha beta gamma delta' );
+		await pressWithModifier( META_KEY, 'Backspace' );
+		await page.keyboard.type( 'delta' );
+		const blockText = await page.evaluate( () => document.activeElement.textContent );
+		expect( blockText ).toBe( 'alpha beta gamma delta' );
+	} );
```
Comment (youknowriad) [inline test/e2e/specs/writing-flow.test.js]: I love these tests, thanks @notnownikki

**Comment by notnownikki** [comment]
I'm going through the linked issues and adding e2e tests where I can. I want to get an e2e test for #6021 next. So far, all is looking good!

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by youknowriad** [comment]
> I'm going through the linked issues and adding e2e tests where I can. I want to get an e2e test for #6021 next. So far, all is looking good!

Nice, feel free to open follow-up PRs with more e2e test 🎉

**Pull request merged**